public/src/client.ts

```diff
@@ -1,72 +1,50 @@
-import { CallOptions, ConnectError, Transport, makeAnyClient } from "@bufbuild/connect";
-import { createAsyncIterable } from "@bufbuild/connect/protocol";
+import { CallOptions, ConnectError, Transport, makeAnyClient } from "@connectrpc/connect";
+import { createAsyncIterable } from "@connectrpc/connect/protocol";
 import {
-    MethodInfo,
-    MethodInfoServerStreaming,
-    MethodInfoUnary,
-    PartialMessage,
-    ServiceType,
+    DescMessage,
+    DescMethodServerStreaming,
+    DescMethodUnary,
+    DescService,
     Message,
-    MethodKind,
-    AnyMessage,
+    MessageInitShape,
+    MessageShape,
 } from "@bufbuild/protobuf";
 
-export type Response<T extends AnyMessage> = {
+export type Response<T extends Message> = {
     error: ConnectError | null
     message: T
 }
 
-/**
- * ResponseClient is a simple client that supports unary and server-streaming
- * methods. Methods will produce a promise for the response message,
- * or an asynchronous iterable of response messages.
- */
-export type ResponseClient<T extends ServiceType> = {
-    [P in keyof T["methods"]]:
-    T["methods"][P] extends MethodInfoUnary<infer I, infer O> ? (request: PartialMessage<I>, options?: CallOptions) => Promise<Response<O>>
-    : T["methods"][P] extends MethodInfoServerStreaming<infer I, infer O> ? (request: PartialMessage<I>, options?: CallOptions) => AsyncIterable<O>
-    : never;
+export type ResponseClient<Desc extends DescService> = {
+    [P in keyof Desc["method"]]: 
+    Desc["method"][P] extends DescMethodUnary<infer I, infer O> ? (request: MessageInitShape<I>, options?: CallOptions) => Promise<Response<MessageShape<O>>> : 
+    Desc["method"][P] extends DescMethodServerStreaming<infer I, infer O> ? (request: MessageInitShape<I>, options?: CallOptions) => AsyncIterable<MessageShape<O>> :
+    never
 };
-
 /**
  * Create a ResponseClient for the given service, invoking RPCs through the
  * given transport.
  */
-export function createResponseClient<T extends ServiceType>(
-    service: T,
-    transport: Transport,
-    errorHandler: (payload?: { method: string; error: ConnectError; } | undefined) => void
-) {
+export function createResponseClient<T extends DescService>(service: T, transport: Transport, errorHandler: (payload?: { method: string; error: ConnectError; } | undefined) => void): ResponseClient<T> {
```

Comment:
This was a very long line. Should we keep the formatting with one parameter per line as before?